kubespawner/spawner.py

```diff
@@ -1992,6 +1992,10 @@ def _load_profile(self, profile_name):
                 self.log.debug(".. overriding KubeSpawner value %s=%s (callable result)", k, v)
             else:
                 self.log.debug(".. overriding KubeSpawner value %s=%s", k, v)
+                if (k == "environment"):
```

Comment:
The lines above have some conditional statements to log what goes on, and ensure that callable functions are evaluated for their value. This logic needs restructured in these. It should be allowed this environment key to be callable as well.

I think:
1. if we have a callable, evaluate it and log doing so in a log.debug
2. if we are merging/updating a dictionary such as we probably want to do in this override, log.debug doing so
3. if we are overriding an existing value, log.debug doing so

Reply:
1) My code now lets the callable do it's job and that sends info to log.debug.  It has no effect on existing values that are not dict so existing log.debuf is sufficient for those.
2) once we find a dict that may need merging, log.debug is sent prior to the attempt
3) If a similar dict for that key exists on spawner, we do shallow copy of kubespawner values over the spawner values.   This is also logged through log.debug.
4) I no longer hardcode environment.  Any similar dict on Kubespawner and Spawner will get the same functionality and logging.